fix the microwave not turning off when the power was cut off

Caldony opened the pull request:
wzhzhzhz

<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
fixes : https://github.com/tgstation/tgstation/issues/51866#issue-646169236

So for some god forsaken reason the microwave would only stop during a power outage if the microwave was already about to fail, but it STILL returned early meaning the microwave  was never told to actually STOP and would never ever turn off unless you unbuilt it.
either way it's fixed now.

## Why It's Good For The Game
wzhzhzhz is good in moderation
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
fix: the microwave will no longer be turned on infinitely when the power is cut off while it's still on
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Caldony committed:
wzhzhzhz

wzhzhzhz

Dennok commented on code/modules/food_and_drinks/kitchen_machinery/microwave.dm:
```diff
@@ -287,8 +287,8 @@
-		if(type == MICROWAVE_PRE)
-			pre_fail()
+		pre_fail()
+		eject()
```
I think, better place this in power_change proc.
https://github.com/tgstation/tgstation/blob/586bc81b4a0326d6b8b35fa30a086496621d01f4/code/modules/power/power.dm#L113-L134
Or use signals.

Caldony committed:
fix the code in power_change()

wzhzhzhz is now fixed by transporting the code in a newly made power_change() proc. the result is the exact same except that BROKEN is now not bundled up with the power fix

Caldony committed:
wzhzhzhzhzhz
